protostuff-parser/src/main/antlr3/io/protostuff/parser/ProtoParser.g

```diff
@@ -113,6 +116,12 @@ annotation_keyval [Proto proto, Annotation annotation]
         )
     ;
 
+doc_entry [Proto proto]
+    :   DOC {
+            proto.addDoc($DOC.text.substring(3).trim());
```

Comment:
On Thu, Apr 16, 2015 at 4:15 PM, Kostiantyn Shchepanovskyi <
notifications@github.com> wrote:

> In protostuff-parser/src/main/antlr3/io/protostuff/parser/ProtoParser.g
> <https://github.com/protostuff/protostuff/pull/119#discussion_r28491146>:
>
> > @@ -113,6 +116,12 @@ annotation_keyval [Proto proto, Annotation annotation]
> >          )
> >      ;
> >
> > +doc_entry [Proto proto]
> > +    :   DOC {
> > +            proto.addDoc($DOC.text.substring(3).trim());
>
> I think it is better to remove .trim(): it removes all whitespaces even
> if we want to keep them.
> For example, code blocks:
>
You cannot have code blocks because the parser terminates on carriage
return (line based docs).
For simplicity, I think its enough.
Later we can add block-based docs where anything inside the start/end
delimiters will be included (including whitespace)

> /// Message description
> /// Usage:
> /// <code>
> /// class X extends Foo<Bar> {
> ///     public static final int BAZ = 1;
> ///     int z() {
> ///         // ...
> ///     }
> /// }
> /// </code>
>
> This way, $DOC.text.substring(3) should be enough.
> Later we can add custom formats like TRIM, REMOVE_LEADING_SPACE.
>
> docs(container) ::= <<
> <if(container.docs)>
>
> /**
> <container.docs:{doc |  * <doc; format="TRIM">}; separator="\n">
>  */
> <endif>
> >>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/protostuff/protostuff/pull/119/files#r28491146>.
>



-- 
When the cat is away, the mouse is alone.
- David Yu